src/font/nerd_font_codegen.py

```diff
@@ -225,23 +236,16 @@ def generate_zig_switch_arms(patch_set):
 
 
 if __name__ == "__main__":
-    path = (
-        Path(__file__).resolve().parent
-        / ".."
-        / ".."
-        / "vendor"
-        / "nerd-fonts"
-        / "font-patcher.py"
-    )
-    with open(path, "r", encoding="utf-8") as f:
-        source = f.read()
+    project_root = Path(__file__).resolve().parents[2]
 
+    patcher_path = project_root / "vendor" / "nerd-fonts" / "font-patcher.py"
+    source = patcher_path.read_text(encoding="utf-8")
```

Comment:
Integrating this into the build system will be easier if it reads the source from stdin.